core/types/block.go

```diff
@@ -1056,6 +1127,18 @@ func (b *Block) BaseFee() *big.Int {
 	}
 	return new(big.Int).Set(b.header.BaseFee)
 }
+func (b *Block) WithdrawalsHash() *common.Hash {
+	if b.header.WithdrawalsHash == nil {
+		return nil
+	}
```

Comment:
seems unnecessary? `b.header.WithdrawalsHash` is already a pointer

Reply:
Or is the goal to return a deep copy? Then need to change `return` part

Reply:
yep good catch, copy/pasta issue, didn't intend to return deep copy. fixed